sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/batch/TransactionalBatchCore.java

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.batch;
+
+import com.azure.cosmos.CosmosAsyncContainer;
+import com.azure.cosmos.CosmosBridgeInternal;
+import com.azure.cosmos.batch.implementation.BatchExecutor;
+import com.azure.cosmos.batch.implementation.ItemBatchOperation;
+import com.azure.cosmos.implementation.OperationType;
+import com.azure.cosmos.implementation.RequestOptions;
+import com.azure.cosmos.models.PartitionKey;
+import reactor.core.publisher.Mono;
+
+import java.util.ArrayList;
+
+import static com.azure.core.util.FluxUtil.withContext;
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkNotNull;
+
+public class TransactionalBatchCore implements TransactionalBatch {
+
+    private final CosmosAsyncContainer container;
+    private final ArrayList<ItemBatchOperation<?>> operations;
+    private final PartitionKey partitionKey;
+    private final String batchSpanName;
+
+    /**
+     * Initializes a new instance of the {@link TransactionalBatchCore} class.
+     *
+     * @param container a container of items on which the batch operations are to be performed.
+     * @param partitionKey the partition key for all items on which batch operations are to be performed.
+     */
+    public TransactionalBatchCore(
+        final CosmosAsyncContainer container,
+        final PartitionKey partitionKey) {
+
+        checkNotNull(container, "expected non-null container");
+        checkNotNull(partitionKey, "expected non-null partitionKey");
+
+        this.container = container;
+        this.operations = new ArrayList<>();
```

Comment:
should this be thread-safe?

Reply:
I can create a synchronized function but is it necessary? TransactionalBatch is one instance, double execution call is expected from user?